crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2304,34 +2326,31 @@ impl<'db> FunctionType<'db> {
     ///
     /// Were this not a salsa query, then the calling query
     /// would depend on the function's AST and rerun for every change in that file.
-    #[salsa::tracked]
-    pub fn return_ty(self, db: &'db dyn Db) -> Type<'db> {
+    #[salsa::tracked(return_ref)]
+    pub fn external_signature(self, db: &'db dyn Db) -> Signature<'db> {
```

Comment:
Should this be the default we use in most places? If so, I would consider calling it `signature`.